tests/test_cli_textconv_with_git.py

```diff
@@ -0,0 +1,256 @@
+            r" version-info std:label \-1 api\.html\#\$ \-\r?\n"
+            r" why std:doc \-1 why\.html Why not…\r?\n"
+            r"\+attrs\.validators\.set_cheat_mode py:function 1 api\.html\#\$ \-\r?\n"
+            r" \r?\n$"
+        )
+
+        # prepare
+        #    project folder
+        path_cwd = scratch_path
+        wd = WorkDir(path_cwd)
+
+        #    On Windows, unresolved executables paths
+        soi_path = "sphobjinv"
+        soi_textconv_path = "sphobjinv-textconv"
+
+        #    On Windows, resolved executables paths
+        resolved_soi_textconv_path = shutil.which(soi_textconv_path)
+        if resolved_soi_textconv_path is None:
+            resolved_soi_textconv_path = soi_textconv_path
+        resolved_soi_path = shutil.which(soi_path)
+        if resolved_soi_path is None:
+            resolved_soi_path = soi_path
+
+        #    git init
+        wd("git init")
+        wd("git config user.email test@example.com")
+        wd('git config user.name "a test"')
+
+        #    inventories: .txt and .inv
+        # scratch_path copied:
+        # objects_attrs.{.txt|.inv.json} --> objects.{.txt|.inv.json}
+        path_cmp = scratch_path / (misc_info.FNames.INIT + misc_info.Extensions.CMP)
+        dst_cmp_path = str(path_cmp)
+
+        path_dec = scratch_path / (misc_info.FNames.INIT + misc_info.Extensions.DEC)
+        dst_dec_path = str(path_dec)
+
+        #    .git/config append
+        path_git_config = path_cwd / ".git" / "config"
+        str_git_config = path_git_config.read_text()
+
+        #    On Windows, RESOLVED path necessary
+        lines = [
+            """[diff "inv"]""",
+            f"""	textconv = {resolved_soi_textconv_path}""",
+        ]
+
+        gc_textconv = sep.join(lines)
+        str_git_config = f"{str_git_config}{sep}{gc_textconv}{sep}"
+        path_git_config.write_text(str_git_config)
+        if platform.system() in ("Linux", "Windows"):
+            print(f"cwd {wd.cwd!s}", file=sys.stderr)
+            print(f"executable path: {resolved_soi_textconv_path}", file=sys.stderr)
+            print(f"""PATHEXT: {os.environ.get("PATHEXT", None)}""", file=sys.stderr)
+            print(f".git/config {str_git_config}", file=sys.stderr)
+
+        #    .gitattributes
+        #    Informs git: .inv are binary files and which cmd converts .inv --> .txt
+        path_ga = path_cwd / ".gitattributes"
+        path_ga.touch()
+        str_gitattributes = path_ga.read_text()
+        ga_textconv = "*.inv binary diff=inv"
+        str_gitattributes = f"{str_gitattributes}{sep}{ga_textconv}"
+        wd.write(".gitattributes", str_gitattributes)
+
+        #    commit (1st)
+        wd.add_command = "git add ."
+        wd.commit_command = "git commit --no-verify --no-gpg-sign -m test-{reason}"
+        wd.add_and_commit(reason="sphobjinv-textconv", signed=False)
+
+        # Act
+        #    make change to .txt inventory (path_dst_dec)
```

Comment:
This is a realistic workflow. A line is added to a .txt --> soi --> .inv

``soi convert -q zlib dec cmp`` with resource absolute paths did not work on Windows. So that issue would not have been found, if had not gone thru the entire workflow.

Will isolate that issue into a separate test and then follow your advice on using two pre-built resources

Reply:
Never mind on this -- I was thinking about it incorrectly. `git diff` acts on the same file over time, not on two different files.

Instead, I'd recommend creating a small, trivial `objects_textconv.inv` for this, and then making a new version and committing a new version of that same `objects_textconv.inv`. That will give the two different versions of the files to be `git diff`-ed.

And then, the `git diff` command in the test would be something like `git diff <sha1>..<sha2> tests/resource/objects_textconv.inv`.

I might still not have this quite right, but it's at least closer.

Reply:
> `soi convert -q zlib dec cmp` with resource absolute paths did not work on Windows. So that issue would not have been found, if had not gone thru the entire workflow.

This doesn't make sense to me, I can use absolute paths on Windows without problems in normal operation mode.

It could be something going funny with the Azure CI environment, but it's concerning that absolute paths aren't working correctly.

Reply:
In the next commit, followed your advice on fixtures and created

- `tests/resource/objects_attrs_plus_one_entry.inv`
- `tests/resource/objects_attrs_plus_one_entry.txt`

`tests/resource/objects_attrs_plus_one_entry.inv` overwrites `cwd/objects.inv`

Reply:
This could be a separate PR. Confirm, on Windows and Azure CI, what's the deal with absolute resource paths and soi convert

In this PR, that code has been removed